modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/portlet/action/EditEntryMVCActionCommand.java

```diff
@@ -364,6 +366,9 @@ private void _sendAddRedirect(
 				redirect, namespace + "classPK", entryId);
 		}
 
+		MultiSessionMessages.clear(actionRequest);
+		MultiSessionMessages.add(actionRequest, "blogsEntryPublished");
+
```

Comment:
This looks odd. Instead of clearing the messages and adding a new one, why not check the condition when `((entry == null) && (workflowAction == ACTION_SAVE_DRAFT))` -- It looks like that's the case that is failing. For example, you could change the "update" conditional to something like:
```
else if ((Validator.isNotNull(redirect) && cmd.equals(UPDATE)) || ((entry == null) && (workflowAction == ACTION_SAVE_DRAFT)) {
... // same logic that you have there
}
```
Not sure if that would work, but we need to discover the case that is failing and make the code explicit about it. Your current solution looks bad because you're overriding things previous code has done, making things a bit messy.